### PR TITLE
New OracleGeometryWriter

### PR DESCRIPTION
**Pull request by hlaebe**
# Complete rewrite of OracleGeometryWriter
- Supports major 2D and 2.5D geometry types
- Uses [odp.net.x86](https://www.nuget.org/packages/odp.net.x86/) NuGet packages for CI build
- Requires Oracle Client installation
- Targets Framework 4.5 minimum

### CODE REVIEW

**Comment by airbreather** [comment]
Sorry about the delay.

I don't currently have a way to test this, so the best I can do is speculate about the Oracle side of things.

I can do a surface-level review of it and comment on the NTS side of things, though.

**Comment by hlaebe** [comment]
No problem.
any comment is welcome. I'm using it internally for a while and finally found some time to di some "polish".  I believe it's not at top level, so I'm really interested in improvements on the NTS side.

TIA

File: NetTopologySuite.IO.Oracle.nuspec [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		</description>
+		<summary>Oracle IO library for NetTopologySuite</summary>
+		<tags>NTS Oracle</tags>
+		<language>en-US</language>
+		<frameworkAssemblies>
+			<frameworkAssembly assemblyName="System" targetFramework="" />
+			<frameworkAssembly assemblyName="System.Core" targetFramework="" />
+		</frameworkAssemblies>
+		<dependencies>
+			<dependency id="NetTopologySuite" version="[1.15.1.0]" />
```
Comment (airbreather) [inline NetTopologySuite.IO.Oracle.nuspec]: This should be `version="1.15.1"`, or perhaps `version="[1.15.1,2.0)"`

File: NetTopologySuite.IO.Oracle/NetTopologySuite.IO.Oracle.csproj [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+    <SolutionDir Condition=" '$(SolutionDir)' == '' ">$(ProjectDir)..\</SolutionDir>
+    <TargetFramework>net45</TargetFramework>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>$(SolutionDir)nts.snk</AssemblyOriginatorKeyFile>
+    <DelaySign>false</DelaySign>
+    <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
+      None
+    </ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
+    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
+    <Authors>Holger Laebe</Authors>
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle/NetTopologySuite.IO.Oracle.csproj]: Agreed

File: NetTopologySuite.IO.Oracle/OracleGeometryReader.cs [hunk 3]
```diff
@@ -1,21 +1,20 @@
-﻿using System;
+﻿#pragma warning disable 1591
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle/OracleGeometryReader.cs]: I am open to this suggestion, I myself have no preferences here.

File: nuget.config [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply nuget.config]: Ok

File: NetTopologySuite.IO.Oracle.nuspec [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+	<metadata>
+		<id>NetTopologySuite.IO.Oracle</id>
+		<version>1.15.1.0</version>
+		<title>NetTopologySuite IO Library for Oracle</title>
+		<authors>Holger Laebe</authors>
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle.nuspec]: Of course you're right, that wasn't an attempt to claim that either. Some entries were automatically preset by the VS extension, please change whatever is necessary.

File: NetTopologySuite.IO.Oracle.nuspec [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+	<metadata>
+		<id>NetTopologySuite.IO.Oracle</id>
+		<version>1.15.1.0</version>
+		<title>NetTopologySuite IO Library for Oracle</title>
+		<authors>Holger Laebe</authors>    
+		<license type="expression">LGPL-2.1-or-later</license>
+		<projectUrl>https://github.com/hlaebe/NetTopologySuite.IO.Oracle</projectUrl>
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle.nuspec]: Agreed

File: NetTopologySuite.IO.Oracle.nuspec [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+	<metadata>
+		<id>NetTopologySuite.IO.Oracle</id>
+		<version>1.15.1.0</version>
+		<title>NetTopologySuite IO Library for Oracle</title>
+		<authors>Holger Laebe</authors>    
+		<license type="expression">LGPL-2.1-or-later</license>
```
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle.nuspec]: The license settings for NuGet are not consistent. I'm not sure if you need to create the license.txt file or an entry in the nuspec file. Unfortunately, I don't know the license under which the extension was installed. Please enter the appropriate license.

File: NetTopologySuite.IO.Oracle.nuspec (hunk 1 above)
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle.nuspec]: Agreed

File: NetTopologySuite.IO.Oracle/NetTopologySuite.IO.Oracle.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (hlaebe) [reply NetTopologySuite.IO.Oracle/NetTopologySuite.IO.Oracle.csproj]: Ok, removed

**Pull request merged**